### PR TITLE
PYMT-1062: Feature wrap flow config

### PR DESCRIPTION
**Pull request by rashmitsingh**
Ticket here: https://loyaltycorp.atlassian.net/browse/PYMT-1062

This PR adds flow config library and exposed a service wrapper to use its functionality.

This is open for a draft review. I still need write a base functional test case to be able to do a real integration test.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#Multi Tenancy
```
Comment (edwardmurrell-loyaltycorp) [inline readme.md:1]: 💯 for having excellent documentation

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+2. Add an XMLDriver for `CodeFoundation\FlowConfig` namespace on your doctrine mapping driver chain. The driver should load maps from `vendor/code-foundation/flow-config/src/Entity/DoctrineMaps` and look for files with extension `.orm.xml` 
+
+3. Implement `LoyaltyCorp\Multitenancy\Services\FlowConfig\Interfaces\FlowConfigurableInterface` on an entity that wishes to be able to write to flow config.
+
+```php
+class User implements LoyaltyCorp\Multitenancy\Services\FlowConfig\Interfaces\FlowConfigurableInterface
+{
+    public function getEntityType(): string
+    {
+        return 'user';
```
Comment (edwardmurrell-loyaltycorp) [inline readme.md:23]: Might be worth adding a note here that this is different for each class.

File: src/Services/FlowConfig/FlowConfig.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+     * @var \CodeFoundation\FlowConfig\Interfaces\ConfigRepositoryInterface
+     */
+    private $flowConfig;
+
+    /**
+     * FlowConfig constructor.
+     *
+     * @param \CodeFoundation\FlowConfig\Interfaces\EntityConfigRepositoryInterface $entityFlowConfig
+     * @param \CodeFoundation\FlowConfig\Interfaces\ConfigRepositoryInterface $flowConfig
+     * @param bool|null $autoFlush
```
Comment (edwardmurrell-loyaltycorp) [inline src/Services/FlowConfig/FlowConfig.php:28]: Good work on seeing that we'll need this. I recommend either adding a note in the code to reference the ticket for autoflush, or a reference to this un-used argument in the ticket.

Comment (rashmitsingh) [reply src/Services/FlowConfig/FlowConfig.php:28]: A mention of why autoFlush is here and suppressed has been added, I will add a note to ticket to remove this suppression when that is done after this gets approved.

**Review by edwardmurrell-loyaltycorp — APPROVED** [review]

**Pull request merged**